Libraries iOS arm build issue "libc++abi.dylib: terminating with uncaught exception"

Multiple PRs are seeing failures in the `iOS arm` job that goes away after a retry.

It looks like an issue with the Mac Hosted Pool machine(s), I have an email thread open with MMS Experts DL.

```
2020-04-02T15:18:18.8662600Z   Commencing build of "native libraries component" for iOS.arm.Debug in /Users/runner/runners/2.165.2/work/1/s/artifacts/obj/native/netcoreapp5.0-iOS-Debug-arm
2020-04-02T15:18:20.0041020Z     Determining projects to restore...
2020-04-02T15:18:20.7508640Z     Restored /Users/runner/runners/2.165.2/work/1/s/eng/empty.csproj (in 425 ms).
2020-04-02T15:18:20.9988160Z   Invoking "/Users/runner/runners/2.165.2/work/1/s/eng/native/gen-buildsys.sh" "/Users/runner/runners/2.165.2/work/1/s/src/libraries/Native/Unix" "/Users/runner/runners/2.165.2/work/1/s/src/libraries/Native/Unix" "/Users/runner/runners/2.165.2/work/1/s/artifacts/obj/native/netcoreapp5.0-iOS-Debug-arm" arm clang "" "" Debug ""  -DCLR_ENG_NATIVE_DIR="/Users/runner/runners/2.165.2/work/1/s/eng/native" -DCMAKE_OSX_SYSROOT=iphoneos -DCMAKE_OSX_DEPLOYMENT_TARGET=7.0 -DCMAKE_OSX_ARCHITECTURES="armv7;armv7s" -DCMAKE_SYSTEM_NAME=iOS
2020-04-02T15:18:22.4717920Z   CMake Warning at /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/Darwin-Initialize.cmake:204 (message):
2020-04-02T15:18:22.4721980Z     No SDK found for architecture 'armv7'
2020-04-02T15:18:22.4723030Z   Call Stack (most recent call first):
2020-04-02T15:18:22.4725570Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/Darwin-Initialize.cmake:213 (_apple_resolve_multi_arch_sysroots)
2020-04-02T15:18:22.4727300Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/iOS-Initialize.cmake:1 (include)
2020-04-02T15:18:22.4728500Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/CMakeSystemSpecificInitialize.cmake:21 (include)
2020-04-02T15:18:22.4751860Z     CMakeLists.txt:4 (project)
2020-04-02T15:18:22.4760680Z  
2020-04-02T15:18:22.4765260Z  
2020-04-02T15:18:22.4770420Z   CMake Warning at /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/Darwin-Initialize.cmake:204 (message):
2020-04-02T15:18:22.4773970Z     No SDK found for architecture 'armv7s'
2020-04-02T15:18:22.4780680Z   Call Stack (most recent call first):
2020-04-02T15:18:22.4782010Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/Darwin-Initialize.cmake:213 (_apple_resolve_multi_arch_sysroots)
2020-04-02T15:18:22.4783530Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/Platform/iOS-Initialize.cmake:1 (include)
2020-04-02T15:18:22.4784440Z     /usr/local/Cellar/cmake/3.17.0_1/share/cmake/Modules/CMakeSystemSpecificInitialize.cmake:21 (include)
2020-04-02T15:18:22.4797310Z     CMakeLists.txt:4 (project)
2020-04-02T15:18:22.4818490Z  
2020-04-02T15:18:22.4846130Z  
2020-04-02T15:18:24.0211860Z   -- The C compiler identification is AppleClang 11.0.0.11000033
2020-04-02T15:18:24.0867500Z   -- Check for working C compiler: /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
2020-04-02T15:18:24.1585660Z   libc++abi.dylib: terminating with uncaught exception of type std::bad_alloc: std::bad_alloc
2020-04-02T15:18:24.1620790Z   /Users/runner/runners/2.165.2/work/1/s/eng/native/gen-buildsys.sh: line 98:  1303 Abort trap: 6           $cmake_command -G "$generator" "-DCMAKE_BUILD_TYPE=$buildtype" "-DCMAKE_INSTALL_PREFIX=$__CMakeBinDir" $cmake_extra_defines $__UnprocessedCMakeArgs -S "$1" -B "$3"
2020-04-02T15:18:24.1698320Z   Failed to generate "native libraries component" build project!
2020-04-02T15:18:24.1703690Z /Users/runner/runners/2.165.2/work/1/s/src/libraries/Native/build-native.proj(36,5): error MSB3073: The command ""/Users/runner/runners/2.165.2/work/1/s/src/libraries/Native/build-native.sh" arm Debug outconfig netcoreapp5.0-iOS-Debug-arm -os iOS " exited with code 134.
2020-04-02T15:18:24.1809270Z ##[error]src/libraries/Native/build-native.proj(36,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command ""/Users/runner/runners/2.165.2/work/1/s/src/libraries/Native/build-native.sh" arm Debug outconfig netcoreapp5.0-iOS-Debug-arm -os iOS " exited with code 134.
2020-04-02T15:18:24.2100790Z
2020-04-02T15:18:24.2107260Z Build FAILED.
```

Official build failures here: https://github.com/dotnet/runtime/issues/34510

--- a/eng/pipelines/runtime-official.yml
+++ b/eng/pipelines/runtime-official.yml
@@ -76,7 +76,7 @@ stages:
       - tvOS_x64
       - tvOS_arm64
       - iOS_x64
-      # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
+      - iOS_arm
       - iOS_arm64
       - OSX_x64
       - Linux_x64
@@ -129,7 +129,7 @@ stages:
       - tvOS_x64
       - tvOS_arm64
       - iOS_x64
-      # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
+      - iOS_arm
       - iOS_arm64
       jobParameters:
         isOfficialBuild: ${{ variables.isOfficialBuild }}
@@ -188,7 +188,7 @@ stages:
       platforms:
       - tvOS_x64
       - tvOS_arm64
-      # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
+      - iOS_arm
       - iOS_arm64
       - iOS_x64
       - Android_arm

--- a/eng/pipelines/runtime.yml
+++ b/eng/pipelines/runtime.yml
@@ -395,7 +395,7 @@ jobs:
     - Android_x64
     - Android_arm
     - tvOS_arm64
-    # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
+    - iOS_arm
     - iOS_x64
     jobParameters:
       liveRuntimeBuildConfig: release
@@ -506,7 +506,7 @@ jobs:
     - Android_x64
     - Android_arm
     - tvOS_arm64
-    # - iOS_arm # https://github.com/dotnet/runtime/issues/34465
+    - iOS_arm
     - iOS_x64
     jobParameters:
       liveRuntimeBuildConfig: release

